File: kb_qa_system/backend/app/schemas/document_chat.py
"""
文档对话相关 Schema

作用：
    定义文档对话功能的请求和响应数据模型。
    用户上传文件后，系统解析文本放入 LLM 上下文，支持针对文档内容提问。
"""

from pydantic import BaseModel, Field
from typing import Optional


class DocumentChatUploadResponse(BaseModel):
    """文档上传响应"""
    session_id: str = Field(..., description="文档对话会话ID，用于后续提问")
    file_name: str = Field(..., description="文件名")
    file_type: str = Field(..., description="文件类型")
    file_size: int = Field(..., description="文件大小（字节）")
    char_count: int = Field(..., description="解析后的文本字符数")
    truncated: bool = Field(False, description="文档是否因过长被截断")


class DocumentChatRequest(BaseModel):
    """文档对话提问请求"""
    session_id: str = Field(..., description="文档对话会话ID")
    question: str = Field(..., min_length=1, max_length=2000, description="问题内容")
    conversation_id: Optional[int] = Field(None, description="对话ID（首次提问可不传，后端自动创建）")


class DocumentFromLibraryRequest(BaseModel):
    """
    从文档库选择文档进行对话的请求

    作用：
        用户选择个人文档库或公共文档库中已处理的文档，
        复用其已清洗的全文内容进行文档对话，无需重新上传和解析。

    字段说明：
        document_id: 文档ID（必须是 status=completed 且当前用户有权访问的文档）
    """
    document_id: int = Field(..., gt=0, description="文档库中的文档ID")

File: kb_qa_system/frontend/src/types/documentChat.ts
/**
 * 文档对话相关类型定义
 *
 * 作用：
 *   定义文档上传、文档对话等相关的 TypeScript 类型，
 *   与后端文档对话接口的响应结构对齐。
 */

/** 文档上传响应（对齐后端 DocumentChatUploadResponse） */
export interface DocumentChatUploadResponse {
  /** 会话ID（用于后续提问） */
  session_id: string;
  /** 文件名 */
  file_name: string;
  /** 文件类型（扩展名，如 .pdf） */
  file_type: string;
  /** 文件大小（字节） */
  file_size: number;
  /** 解析得到的字符数 */
  char_count: number;
  /** 是否被截断（超出最大字符数限制） */
  truncated: boolean;
}

/** 文档对话提问请求（对齐后端 DocumentChatRequest） */
export interface DocumentChatRequest {
  /** 会话ID（由上传响应返回） */
  session_id: string;
  /** 问题内容 */
  question: string;
  /**
   * 对话ID（可选）
   *
   * 作用：
   *   首次提问不传，后端自动创建 Conversation 并在 done 事件返回 conversation_id；
   *   后续追问传入该 ID，使多轮问答归属于同一对话记录（在侧边栏对话历史中显示）。
   */
  conversation_id?: number;
}

/** 文档对话消息（前端展示用） */
export interface DocumentChatMessage {
  /** 角色：user 用户 / assistant AI */
  role: "user" | "assistant";
  /** 消息内容 */
  content: string;
}
